### PR TITLE
WIP: feat(boards)!: refactor common layouts

### PR DESCRIPTION
**Pull request by joelspadin**
Removed the combined .dtsi file for each group of common layouts, and split the position map across the files in each group. This makes reading the position maps slightly more difficult, but it aligns the common layouts with the keyboard-specific ones such as foostan/corne.dtsi in that simply including each header adds the layout, with no need to set status = "okay".

### CODE REVIEW

**Review by petejohanson — COMMENTED** [review]
Ok, seeing this in practice.... I really do like the fact that boards/shields can just explicitly import the layouts they want and run with it. The splitting up of the position map makes it perhaps a tiny bit harder to grok, but also is *cohesive* with everything for a given specific layout in the same file. I'd say we go with this, unless @ReFil has any concerns after their initial work on this.

**Comment by ReFil** [comment]
I like this implementation, I can foresee it saving a lot of confusion when module developers are unsure whether or not to include `status = "okay";` for their specific board

**Comment by joelspadin** [comment]
I _think_ this is ready for review now. I checked that all the position maps make sense by pasting all the files for each layout together into https://zmk-layout-helper.netlify.app/, but I haven't tested that every board builds yet.

**Comment by Nick-Munnich** [comment]
Could you split the corne layout as well?

**Comment by joelspadin** [comment]
> Could you split the corne layout as well?

I didn't notice that there were multiple layouts there before. Done.

**Comment by joelspadin** [comment]
Looks like there are more things for me to update since the commit where I started this branch.

File: app/boards/shields/corne/corne.dtsi [hunk 1]
```diff
@@ -6,7 +6,8 @@
 
 #include <dt-bindings/zmk/matrix_transform.h>
 
-#include <layouts/foostan/corne.dtsi>
+#include <layouts/foostan/corne/5column.dtsi>
+#include <layouts/foostan/corne/6column.dtsi>
```
Comment (caksoylar) [inline app/boards/shields/corne/corne.dtsi:11]: Don't these want to include the position map as well?

**Review by petejohanson — APPROVED** [review]
Thanks!

**Pull request merged**